src/components/structures/auth/Login.tsx

```diff
@@ -392,19 +392,17 @@ export default class LoginComponent extends React.PureComponent<IProps, IState>
                     // look for a flow where we understand all of the steps.
                     const supportedFlows = flows.filter(this.isSupportedFlow);
 
-                    if (supportedFlows.length > 0) {
+                    this.setState({
+                        flows: supportedFlows,
+                    });
+
+                    if (!supportedFlows.length) {
```

Comment:
can we avoid the implicit cast from number to bool, unless you're particularly relying on matching something other than `0` (in which case, please make it clear)


```suggestion
                    if (supportedFlows.length === 0) {
```